src/marten/db/query/set.cr

```diff
@@ -325,6 +325,40 @@ module Marten
           add_query_node(query_node)
         end
 
+        # Returns if a query set includes a specific model.
+        def includes?(value : DB::Model)
```

Comment:
Can we use `M` instead of `DB::Model`? The `Marten::DB::Query::Set` abstraction is a [generic class](https://github.com/martenframework/marten/blob/f0d70d278f814c56777fac7c1317426cc8a8e707/src/marten/db/query/set.cr#L14).

```suggestion
        def includes?(value : M)
```

This will ensure that only `Article` records can be used for `Marten::DB::Query::Set(Article)` queru sets objects for example.